File: mobile/lib/domain/models/steering_state.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'steering_state.freezed.dart';

/// Normalized steering angle snapshot (-1.0 full left .. 1.0 full right).
@freezed
abstract class SteeringState with _$SteeringState {
  const factory SteeringState({
    @Default(0.0) double angle,
  }) = _SteeringState;

  const SteeringState._();

  /// Straight ahead.
  static const centered = SteeringState();
}

File: android/app/src/main/kotlin/dev/fazrigading/wheeldeck/domain/models/SteeringState.kt
package dev.fazrigading.wheeldeck.domain.models

/// Normalized steering angle snapshot (-1.0 full left .. 1.0 full right).
data class SteeringState(
    val angle: Double = 0.0,
) {
    companion object {
        val centered = SteeringState()
    }
}
